### PR TITLE
Update installation.md

### LINKED ISSUE #5712: pip install --upgrade diffusers[torch] fails with Python 3.12 on Windows 11
### Describe the bug

`pip install --upgrade diffusers[torch]`  with Python 3.12 results in a failed installation, with one the (many) errors being: 

`ERROR: Could not find a version that satisfies the requirement torch (from versions: none)`

Workaround: 
Use Python 3.11.

Note:
Writing this to hopefully save others from a day of head scratching.

### Reproduction

Run `pip install --upgrade diffusers[torch]` using Python 3.12

### Logs

_No response_

### System Info

Python 3.12.x

### Who can help?

_No response_

### PR DESCRIPTION
**Pull request by Bhavay-2001**
# What does this PR do?

Fixes #5712.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: docs/source/en/installation.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ You should also install 🤗 Transformers because 🤗 Diffusers relies on its m
 <pt>
 ```bash
 pip install diffusers["torch"] transformers
+Note - PyTorch doesn't support Python 3.12 on Windows. Please use Python 3.11 instead.
```
Comment (stevhliu) [inline docs/source/en/installation.md:44]: This should be placed outside of the code block and preferably before it :)

```suggestion
PyTorch only supports Python 3.8 - 3.11 on Windows.
```

**Review by stevhliu — APPROVED** [review]
Thanks for the update!

File: docs/source/en/installation.md (hunk 1 above)
Comment (Bhavay-2001) [reply docs/source/en/installation.md:44]: Hi @stevhliu, actually @patrickvonplaten suggested to place it inside the the Pytorch framework placeholder. Should I place this before the Pytorch block?

Comment (stevhliu) [reply docs/source/en/installation.md]: Sorry, what I meant is it should be outside of the `bash` code block but inside the `<pt>` one like this:

```md
<pt>

PyTorch only supports Python 3.8 - 3.11 on Windows.

<code snippet here>

</pt>
```

**Comment by stevhliu** [comment]
Hi, we can merge once this [comment](https://github.com/huggingface/diffusers/pull/6438#discussion_r1444994725) is addressed :)

**Comment by Bhavay-2001** [comment]
Sorry I missed that. I will complete this.

**Comment by Bhavay-2001** [comment]
Hi @stevhliu, I have updated the file. Please check once.
Thanks

**Review by stevhliu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**